yarpc/yarpc-go
# Generate code coverage ignore directory info

This PR autogenerates what packages should be ignored for code coverage using the script `scripts/generate-cover-ignore.sh`. This script does the following:

First, calculate the list of ignored packages:

- Ignore all go packages recursively the directories that do not contain yarpc library code:
  - internal/crossdock
  - internal/examples
  - internal/service-test
  - internal/testutils
  - transport/x/cherami/example

- Ignore all go packages that contain a generated go file, except for:
  - api/peer 
  - api/peer/peertest
  - api/middleware/middlewaretest
  - api/transport
  - api/transport/transporttest
  - encoding/thrift
  - encoding/thrift/thriftrw-plugin-yarpc
  - internal/interpolate
  - internal/sync
  - transport/x/redis/redistest

Then:

- Remove all existing `.nocover` files
- Add a `.nocover` file to all ignored directories
- Add all ignored directories to `.codecov.yml`

We may need to adjust the list of directories, but this is a good start. Note that this ends up actually dropping coverage, because `internal/crossdock` was technically being covered to a high level.
